File: ReactTS/src/types/category.ts
export interface Category {
  categoryId: number;
  name: string;
  slug: string;
  displayOrder: number;
  subcategories: Subcategories[];
}

export interface Subcategories {
  subcategoryId: number;
  name: string;
  slug: string;
  displayOrder: number;
  imageUrl: string;
}

File: server/API/Models/DboTables/Category.cs
using API.Database;

namespace API.Models.DboTables;

[DbTable("dbo.category")]
public class Category
{
    [DbPrimaryKey] public int CategoryId { get; set; }

    [DbColumn] public string Name { get; set; } = "";
    
    [DbColumn] public string Slug { get; set; } = "";

    [DbColumn] public int DisplayOrder { get; set; } = 0;

    [DbColumn] public bool IsActive { get; set; } = true;
    
    [DbColumn] public DateTime CreatedAt { get; set; } = DateTime.Now;
    
    [DbColumn] public DateTime? UpdatedAt { get; set; }
}